src/Aspire.Dashboard/Components/Pages/IComponentWithTelemetry.cs

```diff
@@ -0,0 +1,93 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Dashboard.Telemetry;
+
+namespace Aspire.Dashboard.Components.Pages;
+
+/// <summary>
+/// Interface for components that would like to opt-in to telemetry events, capturing
+/// 1) initialization
+/// 2) component properties on each render
+/// 3) component disposal
+/// </summary>
+public interface IComponentWithTelemetry
+{
+    public DashboardTelemetryService TelemetryService { get; }
+    public ComponentTelemetryContext TelemetryContext { get; }
+
+    public void UpdateTelemetryProperties()
+    {
+    }
+}
+
+public static class IComponentWithTelemetryExtensions
+{
+    public static async Task InitializeComponentTelemetryAsync(this IComponentWithTelemetry component)
```

Comment:
What do you think of moving these static methods can be moved onto TelemetryContext? TelemetryContext would need a reference to the service, but that doesn't seem like a big problem.

Reply:
That seems like a good idea, moved